### PR TITLE
Build neutron notebooks in docs

### LINKED ISSUE #963: Make neutron doc pages build automatically using files from Mantid archive
This concerns the neutron data notebooks:
- https://github.com/scipp/scipp/blob/master/docs/tutorials/neutron-diffraction.ipynb
- https://github.com/scipp/scipp/blob/master/docs/tutorials/neutron-data.ipynb
and the instrument view:
- https://github.com/scipp/scipp/blob/master/docs/scipp-neutron/instrument-view.ipynb

This depends on #952 : once the docs build also on Azure, we can download the POWGEN files and run the notebooks so that they do not get outdated.

### PR DESCRIPTION
**Pull request by nvaytet**
Until now, the neutron-related notebooks in the docs were not run for PRs, causing them to constantly be out-of-date.

With this PR, we install `mantid-framework` in the docs build environment and are now able to run the neutron notebooks as part of the builds.

A small python script (`fetch_neutron_data.py`) downloads a limited number of files that are required for the notebooks to run, from the Mantid ftp server.

The instrument view is now interactive in the docs pages! (well only the 3D view, sliders and buttons have no effect)

**Reviewer:** You can check the built pages by downloading the documentation artifact from Azure: https://dev.azure.com/scipp/scipp/_build/results?buildId=456&view=artifacts&type=publishedArtifacts

**Further details:** (in order of appearance)

- A `docs/_templates/layout.html` file was created to expand the width of the docs page div from 800px to 1100px. This makes better use of screen space and was necessary to avoid multiple horizontal scrollbars in the instrument view widgets. An alternative would be to change the width of the instrument view window, but after experimenting I thought it was getting really small, just to fit in the doc pages, while the original fitted nicely in a regular notebook. This is open for discussion.

- The advanced plotting notebook was pretty long and i've split it into smaller pieces: 1-D, 2-D and N-D plotting

- The old `neutron-data` notebook was removed and only the `neutron-diffraction` notebook was kept. I changed its title from `Neutron Data Quick Start` to `Neutron Powder Diffraction`

- A matplotlib warning for a maximum number of opened figures was silenced.

Fixes #963

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
Is this missing an IDF? There are warnings in the docs build on Azure when loading the file about missing detectors. On the other hand at startup it claims "All Instrument definitions up to date."?

We should also disable usage reporting from CI (via the properties file)?

**Comment by SimonHeybrock** [comment]
>     * A `docs/_templates/layout.html` file was created to expand the width of the docs page div from 800px to 1100px.

I still get all the scroll bars on my laptop. Only when zoomong to 80% or less they disappear. That being said, docs look reasonably readable otherwise (CSS seems to avoid scroll bars otherwise, i.e., the 1100px is not hardcoded like in the olden days). What do the other reviewers think about this layout change?

File: docs/scipp-neutron/instrument-view.ipynb [hunk 1]
```diff
@@ -93,14 +108,13 @@
...
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "This works for any file that can be loaded by Mantid that contains a valid instrument description or geometry:"
+    "This works for any file that can be loaded by Mantid that contains a valid instrument description or geometry.\n",
+    "\n",
+    "There are also two different rendering modes:\n",
```
Comment (SimonHeybrock) [inline docs/scipp-neutron/instrument-view.ipynb:113]: This doesn render the list, I think you need a newline here.

**Comment by DanNixon** [comment]
> > ```
> > * A `docs/_templates/layout.html` file was created to expand the width of the docs page div from 800px to 1100px.
> > ```
> 
> I still get all the scroll bars on my laptop. Only when zoomong to 80% or less they disappear. That being said, docs look reasonably readable otherwise (CSS seems to avoid scroll bars otherwise, i.e., the 1100px is not hardcoded like in the olden days). What do the other reviewers think about this layout change?

I'm in the camp that prefers using the entire page width (i.e. `max-width: 100%` in this case), but others will complain that makes the document very wide and difficult to read on a high DPI or widescreen monitor, to which I'd reply "why do you have your browser window that wide in the first place? that is a terrible use of screen space".

I think for the sake of giving things the best chance of being fully visible I'd suggest using the full page width.

The only possible issue are high DPI mobile devices (such as the iPad Pro) which *might* make text too small to easily read and may not handle zooming well. But that isn't a huge concern I don't think.

File: python/tests/compat/test_mantid.py [hunk 2]
```diff
@@ -18,8 +18,9 @@ def mantid_is_available():
         return False
 
 
-@pytest.mark.skipif(not mantid_is_available(),
-                    reason='Mantid framework is unavailable')
+@pytest.mark.skip(reason='Mantid tests are skipped by default, as they '
```
Comment (DanNixon) [inline python/tests/compat/test_mantid.py:21]: Why is this needed? Mantid is not available at the time the tests are run (you only add it to the docs environment, not the build environment).

Comment (nvaytet) [reply python/tests/compat/test_mantid.py:21]: That's a very good point indeed!
Got my environments mixed up there, thanks!

**Comment by nvaytet** [comment]
> > > ```
> > > * A `docs/_templates/layout.html` file was created to expand the width of the docs page div from 800px to 1100px.
> > > ```
> > 
> > 
> > I still get all the scroll bars on my laptop. Only when zoomong to 80% or less they disappear. That being said, docs look reasonably readable otherwise (CSS seems to avoid scroll bars otherwise, i.e., the 1100px is not hardcoded like in the olden days). What do the other reviewers think about this layout change?
> 
> I'm in the camp that prefers using the entire page width (i.e. `max-width: 100%` in this case), but others will complain that makes the document very wide and difficult to read (due to long lines of text) on a high DPI or widescreen monitor, to which I'd reply "why do you have your browser window that wide in the first place? that is a terrible use of screen space".
> 
> I think for the sake of giving things the best chance of being fully visible I'd suggest using the full page width.
> 
> The only possible issue are high DPI mobile devices (such as the iPad Pro) which _might_ make text too small to easily read and may not handle zooming well. But that isn't a huge concern I don't think.

I had tried initially with 100%, but then when I made my browser window large, some stuff looked a little strange, such as very long code cells for small figures. However, I agree that you probably shouldn't be making your window that large in the first place, just use half the screen for the browser window.
I settled for something like 1100px as it made it look pretty close to what my jupyter notebook looks like, and I think it's not a bad idea to try and have the docs look as close as possible to what a user would see in their own notebook. But then when you change your Jupyter browser window size, things also change there (although it seems to change in discrete steps rather than linear scaling).

So I'm fine either way. 100% or 1100px. @SimonHeybrock i'm not sure why you are still seeing scrollbars, does it vary in different browsers?

**Comment by SimonHeybrock** [comment]
> So I'm fine either way. 100% or 1100px. @SimonHeybrock i'm not sure why you are still seeing scrollbars, does it vary in different browsers?

I suppose it is related to some scaling for the retina display? "1100px" scaled for high-dpi display does not fit my screen (13 inch).

**Comment by nvaytet** [comment]
> Is this missing an IDF? There are warnings in the docs build on Azure when loading the file about missing detectors. On the other hand at startup it claims "All Instrument definitions up to date."?
> 
> We should also disable usage reporting from CI (via the properties file)?

Yes I also see the warnings on my own machine, always have. But everything has always appeared to work fine, so I just assumed it was some obsur Mantid warning that I didn't have to care too much about. It would maybe be nice to disable them, but on the other hand it would maybe be useful to have them in the future for debugging purposes if we are getting build errors?

File: docs/scipp-neutron/instrument-view.ipynb (hunk 1 above)
Comment (nvaytet) [reply docs/scipp-neutron/instrument-view.ipynb:113]: ok strange, it does render the list on my local notebook...
